Description: Bug 350 (P2) - Restore focus to TrackPanel when play/rec is hit and
 the focus is in the device toolbar.
Origin: upstream, http://code.google.com/p/audacity/source/detail?r=11101

--- a/src/toolbars/DeviceToolBar.cpp
+++ b/src/toolbars/DeviceToolBar.cpp
@@ -324,6 +324,18 @@
    if (gAudioIO) {
       // we allow changes when monitoring, but not when recording
       bool audioStreamActive = gAudioIO->IsStreamActive() && !gAudioIO->IsMonitoring();
+      
+      // Here we should relinquish focus
+      if (audioStreamActive) {
+         wxWindow *focus = wxWindow::FindFocus(); 
+         if (focus == mHost || focus == mInput || focus == mOutput || focus == mInputChannels) {
+            AudacityProject *activeProject = GetActiveProject();
+            if (activeProject) {
+               activeProject->GetTrackPanel()->SetFocus();
+            }
+         }
+      }
+      
       mHost->Enable(!audioStreamActive);
       mInput->Enable(!audioStreamActive);
       mOutput->Enable(!audioStreamActive);
